pandas/core/dtypes/concat.py

```diff
@@ -132,6 +204,75 @@ def is_nonempty(x) -> bool:
     return np.concatenate(to_concat, axis=axis)
 
 
+def concat_arrays(to_concat):
+    """
+    Alternative for concat_compat but specialized for use in the ArrayManager.
+
+    Differences: only deals with 1D arrays (no axis keyword) and does not skip
+    empty arrays to determine the dtype.
+    In addition ensures that all NullArrayProxies get replaced with actual
```

Comment:
could NullArrayProxy be used to handle sketchy cases like [this](https://github.com/pandas-dev/pandas/pull/39574/files#diff-0cece63708dac34cbcecb01dabc98083070a9cfbdf7c69d24722e8c363fc39d7R347)?

Reply:
That specific test is still skipped here because it's a special case of a DataFrame being appended with a non-aligned Series (in which case the reindexing happens inside the DataFrame.append, instead of inside the concat machinery which knows better how to handle this), but once that is fixed, then yes, this will properly preserve datetime dtype for that test case.

Reply:
i was referring to using this in the non-ArrayManager version, but i think that may no longer be necessary following the most recent updates to #39574